web/ui/module/codemirror-promql/src/parser/parser.ts

```diff
@@ -117,28 +110,28 @@ export class Parser {
         this.checkCallFunction(node);
         break;
       case ParenExpr:
-        this.checkAST(walkThrough(node, Expr));
+        this.checkAST(node.getChild('Expr'));
         break;
       case UnaryExpr:
-        const unaryExprType = this.checkAST(walkThrough(node, Expr));
+        const unaryExprType = this.checkAST(node.getChild('Expr'));
         if (unaryExprType !== ValueType.scalar && unaryExprType !== ValueType.vector) {
           this.addDiagnostic(node, `unary expression only allowed on expressions of type scalar or instant vector, got ${unaryExprType}`);
         }
         break;
       case SubqueryExpr:
-        const subQueryExprType = this.checkAST(walkThrough(node, Expr));
+        const subQueryExprType = this.checkAST(node.getChild('Expr'));
         if (subQueryExprType !== ValueType.vector) {
           this.addDiagnostic(node, `subquery is only allowed on instant vector, got ${subQueryExprType} in ${node.name} instead`);
         }
         break;
       case MatrixSelector:
-        this.checkAST(walkThrough(node, Expr));
+        this.checkAST(node.getChild('Expr'));
         break;
       case VectorSelector:
         this.checkVectorSelector(node);
         break;
       case StepInvariantExpr:
-        const exprValue = this.checkAST(walkThrough(node, Expr));
+        const exprValue = this.checkAST(node.getChild('Expr'));
```

Comment:
isn't possible to pass the node ID instead ? 

I would prefer to keep the usage of the constant `Expr`

Reply:
Node groups (see [here](https://lezer.codemirror.net/docs/ref/#common.NodeProp^group)) are identified only by strings.

Reply:
ah yes I didn't pay enough attention to the changes in the grammar.
Thank you for the explanation :)